libtiledbsoma/src/soma/soma_column.cc

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file   soma_column.cc
+ *
+ * @section LICENSE
+ *
+ * The MIT License
+ *
+ * @copyright Copyright (c) 2024 TileDB, Inc.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ *
+ * @section DESCRIPTION
+ *
+ *   This file defines the SOMAColumn class.
+ */
+
+#include "soma_column.h"
+
+namespace tiledbsoma {
+
+template <>
+std::pair<std::string, std::string> SOMAColumn::core_domain_slot<std::string>()
+    const {
+    return std::pair<std::string, std::string>("", "");
+}
+
+template <>
+std::pair<std::string, std::string>
+SOMAColumn::core_current_domain_slot<std::string>(
+    const SOMAContext& ctx, Array& array) const {
+    try {
+        std::pair<std::string, std::string>
+            current_domain = std::any_cast<std::pair<std::string, std::string>>(
+                _core_current_domain_slot(ctx, array));
+
+        if (current_domain.first == "" && (current_domain.second == "\x7f" ||
```

Comment:
There was a comment here about why both `\x7f` and `\xff` -- it needs to be retained. This is very confusing without that comment. (Which is why I put the comment in.)

https://github.com/single-cell-data/TileDB-SOMA/blob/a8349a5cdd295bdb04b965e3e91c3a8959e044e9/libtiledbsoma/src/soma/soma_array.h#L901-L920